c10/core/CPUCachingAllocator.cpp

```diff
@@ -27,24 +22,21 @@ inline void* CPUCachingAllocator::allocate_and_cache(const size_t bytes) {
     // again.
     // Try again.
     ptr = c10::alloc_cpu(bytes);
+    TORCH_CHECK(ptr, "Memory allocation failed for ", bytes, " bytes.");
```

Comment:
Wait, does alloc_cpu throw or return null on failure?  It seems like you are looking for both, but only checking null on this path.

Reply:
My bad. The reason I had it and try..catch was because alloc_cpu was throwing and then I forgot about it and though, OMG I am not checking for valid memory. Thanks for the catch.